lib/moonshot/stack.rb

```diff
@@ -195,6 +198,24 @@ def load_parameters_file
       end
     end
 
+    # @return [String] the path to the parameters file without extension.
+    def raw_template_file_name
+      @raw_template_file_name ||= File.join(Dir.pwd, 'cloud_formation', @app_name)
+    end
+
+    def load_template_file
+      json_template = JsonStackTemplate.new("#{raw_template_file_name}.json", log: @log)
+      yaml_template = YamlStackTemplate.new("#{raw_template_file_name}.y*ml", log: @log)
+      case
+      when json_template.exist?
+        json_template
+      when yaml_template.exist?
+        yaml_template
+      else
+        raise
```

Comment:
RE: your question about exceptions. I generally avoid making new exception classes unless I actually expect someone would use them to implement recovery logic. In this case, `RuntimeError` seems appropriate, because we're just going to rewrite this to a Thor::Error in the CLI class.